install/kubernetes/cilium/charts/operator/templates/deployment.yaml

```diff
@@ -33,7 +33,13 @@ spec:
         - --config-dir=/tmp/cilium/config-map
         - --debug=$(CILIUM_DEBUG)
         command:
-        - cilium-operator
+{{- if .Values.global.eni }}
+        - cilium-operator-aws
+{{- else if .Values.global.azure.enabled }}
+        - cilium-operator-azure
+{{- else }}
+        - cilium-operator-generic
```

Comment:
This will imply `--ipam=cluster-pool` will be the default. Is that reasonable for new installations, or we should stick to one of the older modes for some time?